src/atc/transformers/select_and_cast_columns_transformer_nc.py

```diff
@@ -0,0 +1,41 @@
+from typing import List, Union
+
+import pyspark.sql.functions as F
+import pyspark.sql.types as T
+from pyspark.sql import DataFrame
+
+from atc.etl import TransformerNC
+
+
+class SelectAndCastColumnsTransformerNC(TransformerNC):
```

Comment:
Please consider generalizing, see my comment above

Reply:
I have added the ability for case-insensitive matching and aligned with simple_sql_transformer.

I case we in future add the sec trunc feature to the sql loader directly, the simple_sql_transformer could be deprecated. I will make an issue on this.